src/app/page.tsx

```diff
@@ -1,10 +1,23 @@
+import { getServerSession } from "next-auth/next";
+import { getProviders } from "next-auth/react";
+import Link from "next/link";
+import { getServerAuthSession } from "~/server/auth";
 
-
-export default async function Home() {
+const Home = async (): Promise<React.JSX.Element> => {
+  const session = await getServerAuthSession();
+  console.log("session ", session);
   return (
     <main>
-      <div>HI</div>
+      <div>
+        <Link
+          href={session ? "/api/auth/signout" : "/api/auth/signin"}
+          className="rounded-full bg-white/10 px-10 py-3 font-semibold no-underline transition hover:bg-white/20"
+        >
```

Comment:
Not sure if `Link` works here. Probably should be a `Button` that calls corresponding method